prov/coll/src/coll_cq.c

```diff
@@ -68,18 +68,32 @@ static struct fi_ops_cq coll_cq_ops = {
 
 int coll_cq_open(struct fid_domain *domain, struct fi_cq_attr *attr,
 		 struct fid_cq **cq_fid, void *context)
+{
+	return coll_cq_init(domain, attr, cq_fid, &ofi_cq_progress, context);
+}
+
+int coll_cq_init(struct fid_domain *domain,
+		struct fi_cq_attr *attr, struct fid_cq **cq_fid,
+		ofi_cq_progress_func progress, void *context)
```

Comment:
I *think* these changes are creating some common code for a peer cq, and aren't specifically tied to being used by a collective focused provider.  I would look at starting down the path of creating a new include/ofi_peer.h file, along with a prov/peer directory where some common peer provider related structures and definitions can reside.

This function, along with some of the base ops would be moved there.

Reply:
@shefty, please take a look at #8302